crates/metaslang/cst/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+pub mod cst;
+pub mod cursor;
+pub mod query;
+pub mod text_index;
+
+pub trait Kind:
+    Sized
+    + Copy
+    + Clone
+    + PartialEq
+    + Eq
+    + std::fmt::Display
+    + std::fmt::Debug
+    + serde::Serialize
+    + for<'a> std::convert::TryFrom<&'a str, Error = strum::ParseError>
+{
+}
+
+impl<T> Kind for T where
+    T: Sized
+        + Copy
+        + Clone
+        + PartialEq
+        + Eq
+        + std::fmt::Display
+        + std::fmt::Debug
+        + serde::Serialize
+        + for<'a> std::convert::TryFrom<&'a str, Error = strum::ParseError>
+{
+}
+
+pub trait TerminalKind: Kind {
+    fn is_trivia(&self) -> bool;
+}
+pub trait NonTerminalKind: Kind {}
+pub trait EdgeKind: Kind {}
+
+pub trait KindTypes: Clone + PartialEq {
+    type NonTerminalKind: NonTerminalKind;
+    type TerminalKind: TerminalKind;
+    type EdgeKind: EdgeKind;
```

Comment:
Not sure I understand why "label" is renamed to "edge" here, as opposed to the rest of the codebase.

Reply:
Because that is the more correct term for trees/graphs

Reply:
Agreed, but I think it is going to be problematic to have both terminologies used here (and exposed to end users as well).
If so, we should probably look into using "edges" everywhere in #834 .. thoughts?

Reply:
yes, we should use `Edge` everywhere.